torch/_inductor/codegen/common.py

```diff
@@ -2024,7 +2025,24 @@ def do_cse(v):
                             bounds=bounds,
                             dtype=output_dtype,
                         )
+
+                        nonlocal idx
+                        if (
+                            config.test_configs.runtime_triton_dtype_assert
+                            and V.graph.get_current_device_or_throw().type == "cuda"
```

Comment:
Why only cuda? Does this break on non-GPU triton backends like CPU or MTIA?

Reply:
It can work for CPU-triton or mtia. I was trying to avoid cpu cpp codegen. I can update to be more generic.

Reply:
Makes sense! I guess we could just exclude that specific backend then?